geomstats/geometry/pre_shape.py

```diff
@@ -0,0 +1,405 @@
+"""Kendall Pre-Shape space."""
+
+import logging
+
+import geomstats.backend as gs
+from geomstats.algebra_utils import from_vector_to_diagonal_matrix
+from geomstats.errors import check_tf_error
+from geomstats.geometry.embedded_manifold import EmbeddedManifold
+from geomstats.geometry.hypersphere import Hypersphere
+from geomstats.geometry.matrices import Matrices, MatricesMetric
+from geomstats.geometry.riemannian_metric import RiemannianMetric
+
+TOLERANCE = 1e-6
+
+
+class PreShapeSpace(EmbeddedManifold):
+    """Class for the Kendall pre-shape space.
+
+    The pre-shape space is the sphere of the space of centered k-ad of
+    landmarks in R^m (for the Frobenius norm). It is endowed with the
+    spherical Procrustes metric d(x, y):= arccos(tr(xy^t)).
+
+    Parameters
+    ----------
+    k_landmarks : int
+        Number of landmarks
+    m_ambient : int
+        Number of coordinates of each landmark.
+    """
+
+    def __init__(self, k_landmarks, m_ambient):
+        super(PreShapeSpace, self).__init__(
+            dim=m_ambient * (k_landmarks - 1) - 1,
+            embedding_manifold=Matrices(m_ambient, k_landmarks))
+        self.embedding_metric = self.embedding_manifold.metric
+        self.k_landmarks = k_landmarks
+        self.m_ambient = m_ambient
+        self.metric = ProcrustesMetric(k_landmarks, m_ambient)
+
+    def belongs(self, point, atol=TOLERANCE):
+        """Test if a point belongs to the pre-shape space.
+
+        This tests whether the point is centered and whether the point's
+        Frobenius norm is 1.
+
+        Parameters
+        ----------
+        atol
+        point : array-like, shape=[..., m_ambient, k_landmarks]
```

Comment:
Yes in product manifolds and landmarks spaces, it is `[..., k_landmarks, m_ambient]` but @emaignant wanted to be consistent with Kendall's convention and the representation of vectors as columns...

Reply:
Yes I understand (also saw the `k`, `m` naming to be consistent with Kendall's convention) , but I still think we should be consistent in our codebase, where dimension of the space comes at the end :/ We could add remarks in several docstring to point out that we change the Kendall convention from `m_ambient, k_landmarks` to `k_landmarks, m_ambient` so that the users are fully aware?

Reply:
Ok I made the change and added a remark in the main docstring